Assets/HoloToolkit/SharingWithUNET/Scripts/UNetAnchorManager.cs

```diff
@@ -0,0 +1,623 @@
+        /// </summary>
+        private GameObject objectToAnchor;
+
+        /// <summary>
+        /// Sometimes we'll see a really small anchor blob get generated.
+        /// These tend to not work, so we have a minimum trustworthy size.
+        /// </summary>
+        private const uint MinTrustworthySerializedAnchorDataSize = 500000;
+
+        /// <summary>
+        /// List of bytes that represent the anchor data to export.
+        /// </summary>
+        private List<byte> exportingAnchorBytes = new List<byte>(0);
+
+        /// <summary>
+        /// Keeps track of if we created the anchor.
+        /// </summary>
+        private bool createdAnchor = false;
+
+        /// <summary>
+        /// Previous anchor name.
+        /// </summary>
+        private string oldAnchorName = "";
+
+        /// <summary>
+        /// The anchorData to import.
+        /// </summary>
+        private byte[] anchorData = null;
+
+        /// <summary>
+        /// Tracks if we have updated data to import.
+        /// </summary>
+        private bool gotOne = false;
+
+        /// <summary>
+        /// Keeps track of the name of the anchor we are exporting.
+        /// </summary>
+        private string exportingAnchorName;
+#endif
+
+        /// <summary>
+        /// Ensures that the scene has what we need to continue.
+        /// </summary>
+        /// <returns>True if we can proceed, false otherwise.</returns>
+        private bool CheckConfiguration()
+        {
+            networkTransmitter = GenericNetworkTransmitter.Instance;
+            if (networkTransmitter == null)
+            {
+                Debug.Log("No UNetNetworkTransmitter found in scene");
+                return false;
+            }
+
+            networkManager = NetworkManager.singleton;
+            if (networkManager == null)
+            {
+                Debug.Log("No NetworkManager found in scene");
+                return false;
+            }
+
+            if (SharedCollection.Instance == null)
+            {
+                Debug.Log("No SharedCollection found in scene");
+                return false;
+            }
+
+#if UNITY_WSA
+            objectToAnchor = SharedCollection.Instance.gameObject;
+
+            spatialMapping = SpatialMappingManager.Instance;
+            if (spatialMapping == null)
+            {
+                Debug.Log("Spatial mapping not found in scene. Better anchor locations can be found if a SpatialMappingManager is in the scene");
+            }
+#endif
+
+            return true;
+        }
+
+        private void Start()
+        {
+            if (!CheckConfiguration())
+            {
+                Debug.Log("Missing required component for UNetAnchorManager");
+                Destroy(this);
+                return;
+            }
+
+#if UNITY_WSA
+#if UNITY_2017_2_OR_NEWER
+            if (HolographicSettings.IsDisplayOpaque)
+#else
+            if (!VRDevice.isPresent && !Application.isEditor)
```

Comment:
I can't seem to wrap my head around what this is checking for. No headset and not the editor. What case is that?

Reply:
Ahh I think we want to make sure the headset _is_ present here. Good catch.

Reply:
I believe `VRDevice.isPresent` returns true on HoloLens.

Reply:
Oh wait...I misread your statement. Makes sense now!